packages/service/ni_measurement_plugin_sdk_service/_internal/parameter/metadata.py

```diff
@@ -131,6 +131,8 @@ def _validate_default_value_type(parameter_metadata: ParameterMetadata) -> None:
             expected_element_type = type(
                 get_type_default(parameter_metadata.type, False, type(default_value[0]))
             )
+        else:
+            expected_element_type = expected_type
```

Comment:
Should I create a new instance of `expected_type` for `expected_element_type`?

Reply:
No, see https://github.com/ni/measurement-plugin-python/pull/832#discussion_r1708059094